projects/igniteui-angular/src/lib/services/overlay/overlay.ts

```diff
@@ -308,23 +309,25 @@ export class IgxOverlayService implements OnDestroy {
      * Note created instance is in root scope, prefer the `viewContainerRef` overload when local injection context is needed.
      *
      * @param component Component Type to show in overlay
-     * @param settings Display settings for the overlay, such as positioning and scroll/close behavior.
+     * @param settings (optional): Create settings for the overlay, such as positioning and scroll/close behavior.
+     * Includes also an optional `Injector` to add to the created dynamic component's injectors.
      * @returns Id of the created overlay. Valid until `detach` is called.
      */
-    public attach(component: Type<any>, settings?: OverlaySettings): string;
+    public attach(component: Type<any>, settings?: OverlayCreateSettings): string;
+    // TODO: change third parameter to OverlayCreateSettings and allow passing of Injector and so on.
     /**
      * Generates an Id. Provide this Id when calling the `show(id)` method
      *
      * @param component Component Type to show in overlay
      * @param viewContainerRef Reference to the container where created component's host view will be inserted
-     * @param settings Display settings for the overlay, such as positioning and scroll/close behavior.
+     * @param settings (optional): Display settings for the overlay, such as positioning and scroll/close behavior.
      */
     public attach(component: Type<any>, viewContainerRef: ViewContainerRef, settings?: OverlaySettings): string;
     public attach(
         componentOrElement: ElementRef | Type<any>,
-        viewContainerRefOrSettings?: ViewContainerRef | OverlaySettings,
-        moduleRefOrSettings?: OverlaySettings): string {
-        const info: OverlayInfo = this.getOverlayInfo(componentOrElement, viewContainerRefOrSettings);
+        viewContainerRefOrSettingsOrCreate?: ViewContainerRef | OverlaySettings | OverlayCreateSettings,
```

Comment:
this is still viewContainerRefOrSettings (regardless of the type of settings imo).
You could even use the extended interface here because those should be compatible, no?

Reply:
Same for `getOverlayInfo` if you have to pass the overload variants there too, though I kind of imagined those moving out to attach and that being `createComponet` only